### ISSUE TITLE
[BUG] Dependency on `reactor-netty-http:1.x.x` causes io.netty.resolver.dns.macos.MacOSDnsServerAddressStreamProvider in Apple Silicon

### ISSUE THREAD

**Issue by SayakMukhopadhyay** [Azure/azure-sdk-for-java#32979]
**Describe the bug**
Azure SDK has dependency on `reactor-netty-http:1.x.x` which includes only x86-64 binaries of the `netty-resolver-dns`. This causes the DNS resolver to not load in Apple Silicon based systems. The bug has been reported in https://github.com/reactor/reactor-netty/issues/2440 and the information and discussion there would be more useful than anything I can provide here.

Right now, the workaround is ugly and is described [here](https://github.com/netty/netty/issues/11020#issuecomment-1193280582).

**Expected behavior**
We should not need to use terrible workarounds to use the azure SDK.

**Additional context**
Although this is an upstream issue, I am raising this issue here since reactor-netty has already made a fix at https://github.com/reactor/reactor-netty/pull/2448 but that will only be published in the 2.x.x branch, although the 2.0.0 release is still in the milestone phase. I am raising this issue so the this project has enough time and scope to move to 2.0.0 by the time it releases.

**Comment by alzimmermsft** [comment]
Thank you for filing this @SayakMukhopadhyay.

I'll read through the linked GitHub issues over at Reactor Netty and get back to this with options for the Azure SDKs for Java.

**Comment by alzimmermsft** [comment]
Okay, was able to read through the issues that were linked out from this and wanted to make sure I'm understanding correctly. The issue isn't directly caused by the Azure SDKs but transitively through Reactor Netty due to their dependency on the macOs x86-64 native DNS dependency. Based on that my guess is the work around is either including both the native x86-64 and aarch64 or excluding the x86-64 dependency using Maven excludes and including the aarch64 dependency within an application, does that sound right?

**Comment by SayakMukhopadhyay** [comment]
Yep, your analysis is correct. This is not an issue with the SDK itself. But I feel that the problem with the transitive dependency is important enough to track when the fix has been released upstream and to incorporate the fixed version in this SDK. Hence I created the issue.

**Comment by alzimmermsft** [comment]
Doubly thanks for filing this @SayakMukhopadhyay!

We'll keep an eye on how Reactor Netty/Netty progresses with the root issue and make sure to update the SDKs when this is available. And if we think of anything we can do on our side to make this easier we'll update this further.

@srnagar @JonathanGiles FYI, don't know if this is anything that can be encompassed in tooling
